fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -632,4 +669,435 @@ public long getSecond() {
     public int hashCode() {
         return 31 * super.hashCode() + Objects.hashCode(unixTimestamp(TimeZone.getDefault()));
     }
+
+    // parset the date string value in 'value' by 'format' pattern.
+    // return the next position to parse if hasSubVal is true.
+    // throw InvalidFormatException if encounter errors.
+    // this method is exaclty same as from_date_format_str() in be/src/runtime/datetime_value.cpp
+    // change this method should also change that.
+    public int fromDateFormatStr(String format, String value, boolean hasSubVal) throws InvalidFormatException {
```

Comment:
Move these util methods to a single util class?

Reply:
It is now only used for DateLiteral, and this method is also in `datetime_value.cpp` in BE.
So I think we can just leave it here.

Reply:
Ok. I think the Literal class should be simple. But which isn't a big issue.